src/Symfony/Component/Notifier/Bridge/LightSms/LightSmsTransport.php

```diff
@@ -0,0 +1,172 @@
+        '38' => 'Phone number already exists in this database',
+        '39' => 'Phone number does not exist in this database',
+    ];
+
+    public function __construct(string $login, string $password, string $from, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->login = $login;
+        $this->password = $password;
+        $this->from = $from;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('lightsms://%s/external/get/send.php?from=%s', $this->getEndpoint(), $this->from);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $timestamp = time();
+
+        $signature = $this->generateSignature($message, $timestamp);
+
+        $endpoint = sprintf(
+            'https://www.%s/external/get/send.php?login=%s&signature=%s&phone=%s&text=%s&sender=%s&timestamp=%s',
+            $this->getEndpoint(),
+            $this->login,
+            $signature,
+            $this->escapePhoneNumber($message->getPhone()),
+            $this->escapeSubject($message->getSubject()),
+            $this->from,
+            $timestamp
+        );
+
+        $response = $this->client->request('GET', $endpoint);
+
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
+            throw new TransportException('Unable to send the SMS: ', $response);
+        }
+
+        $content = $response->toArray(false);
+
+        if (isset($content[''])) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$content['']['error']], $response);
+        }
+
+        if (isset($content['error'])) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$content['error']], $response);
+        }
+
+        $phone = $this->escapePhoneNumber($message->getPhone());
+        if (32 === $content[$phone]['error']) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$content['error']], $response);
+        }
+
+        if (0 == $content[$phone]['error']) {
+            $sentMessage = new SentMessage($message, (string) $this);
+            $sentMessage->setMessageId($content[$phone]['id_sms']);
+        }
+
+        return $sentMessage;
+    }
+
+    private function generateSignature(SmsMessage $message, string $timestamp): string
+    {
+        $params = [
+            'timestamp' => $timestamp,
+            'login' => $this->login,
+            'phone' => $this->escapePhoneNumber($message->getPhone()),
+            'sender' => $this->from,
+            'text' => $this->escapeSubject($message->getSubject()),
```

Comment:
you could pass all this parameters to the `getSignature` function to prevent computing such things twice.

Reply:
Yes. Changed logic.